### PR TITLE
Adds Agency User / Fisher registration page

### PR DESCRIPTION
**Pull request by fermion**
Adds a registration page for fishers or agency users. Should be mostly complete except for wording, etc., we'll want to have @inlovewithwhales2020 take a look and come up with the wording that they're happy with. Some GIFs

#### Requesting Fisher access
smelts-fisher

#### Requesting Agency User access
smelts-agency-user

#### Form validations
fisher-errors

#### Errors updated specifically, after creating the above GIFs
Screen Shot 2021-01-12 at 21 15 32

Note the form validations look like poop. I figure we'll also need guidance there @inlovewithwhales2020.

/cc @AlbertGrimley for 👀

### CODE REVIEW

File: app/views/users/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          <% if resource.errors[:access_needed].any? %>
+            <p class="note error" id="username-input-validation" data-show-if-blank>
+              Please choose if you need access as a Fisher or as an Agency User.
+            </p>
+          <% end %>
+          <p class="note" data-show-if="AgencyUser" hidden>
+            This is for users from federal agencies interested in using the Smelts trap location data for research purposes.
+          </p>
+          <p class="note" data-show-if="Fisher" hidden>
+            As a Fisher you will be able to upload data about your traps and their locations.
```
Comment (AlbertGrimley) [inline app/views/users/registrations/new.html.erb:32]: As a Fisher, you will be able to upload data about your traps and their locations, view your traps, and the locations of nearby traps.

Comment (fermion) [reply app/views/users/registrations/new.html.erb]: https://github.com/smelts-tech/ropeless_web/pull/22/commits/7287b497c6a28eee68272fdf0dda90a6a985becd

**Review by AlbertGrimley — APPROVED** [review]

**Comment by inlovewithwhales2020** [comment]
@fermion -- Nice work! SO AWESOME! And a couple of quick comments -- 

- For the address, the user needs to include their town, state and zip code and they need to be separate fields
- When you say "form validations", I'm thinking it's the pink boxes? I'm actually ok with pink :) -- I will work on the wording today.

**Comment by fermion** [comment]
@inlovewithwhales2020 in hindsight that sure makes a lot of sense 😆 

Would you mind adding an issue to capture adding those fields and requiring them for Fishers who are requesting access?

**Comment by inlovewithwhales2020** [comment]
done! it's issue #30

**Pull request merged**